tcms/core/ajax.py

```diff
@@ -194,105 +191,79 @@ class Objects(object):
         __all__ = ['plan', 'case', 'run']
 
         def __init__(self, request, template_name):
-            self.request = request
             self.template_name = template_name
-            for o in self.__all__:
-                if request.GET.get(o):
-                    self.object = o
-                    self.object_pks = request.GET.getlist(o)
-                    break
+            self._get_object_from_request(request)
+
+        def _get_object_from_request(self, request):
+            for obj in self.__all__:
+                if request.GET.get(obj):
+                    self.object = obj
+                    self.object_pk = request.get(obj)
+                    return
 
         def get(self):
             func = getattr(self, self.object)
             return func()
 
         def plan(self):
-            return self.template_name, TestPlan.objects.filter(
-                pk__in=self.object_pks)
+            return self.template_name, TestPlan.objects.get(
+                pk=self.object_pk)
 
         def case(self):
-            return self.template_name, get_selected_testcases(self.request)
+            return self.template_name, TestCase.objects.get(
+                pk=self.object_pk)
 
         def run(self):
-            self.template_name = 'run/get_tag.html'
-            return self.template_name, TestRun.objects.filter(
-                pk__in=self.object_pks)
+            return 'run/get_tag.html', TestRun.objects.get(
+                pk=self.object_pk)
 
     class TagActions(object):
         __all__ = ['add', 'remove']
 
         def __init__(self, obj, tag):
             self.obj = obj
-            self.tag = string_to_list(tag)
-            self.request = request
+            self.tag = tag
 
         def add(self):
-            for tag_str in self.tag:
-                try:
-                    tag, c = Tag.objects.get_or_create(name=tag_str)
-                    for o in self.obj:
-                        o.add_tag(tag)
-                except IntegrityError:
-                    return "Error when adding %s " \
-                           "Tag with this key already exists in the database." % self.tag
-                except Exception:
-                    return "Error when adding %s" % self.tag
+            tag, c = Tag.objects.get_or_create(name=self.tag)
```

Comment:
Use `_` instead of `c`. Underscore is a special variable, meaning we don't care about the result.